### PR TITLE
Feat/drag

### PR DESCRIPTION
**Pull request by purpose233**
- support rect mark interaction

### CODE REVIEW

File: src/editor/interaction/resize-controller.ts [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      this.resizeEndListeners.forEach(listener => listener.call(null, event, nextRect));
+      this.isDragging = false;
+    }
+    event.stopPropagation();
+  };
+
+  private computeRect(point: IPointLike) {
+    const nextRect = Object.assign({}, this.startRect);
+    // left
+    if (this.resizeMode & 0b10000) {
```
Comment (hunmix) [inline src/editor/interaction/resize-controller.ts:167]: That's not very clear, how about a constant instead?
like:
```
const LEFT = '0b10000'
if (this.resizeMode & LEFT)
```

File: src/editor/marks/rect.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import type { IView } from '@visactor/vgrammar';
+import { elementVisualConfig } from '../../config/visual';
+import { BaseMark } from './base';
+import type { RectMarkVisual } from '../../typings/mark';
+
+export class RectMark extends BaseMark {
+  readonly type: 'rect';
+  constructor(view: IView, visualConfig: RectMarkVisual) {
+    super(view, visualConfig);
+    this.defaultVisual = elementVisualConfig.text;
```
Comment (hunmix) [inline src/editor/marks/rect.ts]: wrong type of config

File: src/editor/interaction/resize-controller.ts (hunk 1 above)
Comment (purpose233) [reply src/editor/interaction/resize-controller.ts:167]: Rejected. Next one.

File: src/editor/marks/rect.ts (hunk 2 above)
Comment (purpose233) [reply src/editor/marks/rect.ts]: Fixed.

**Pull request merged**